trino/client.py

```diff
@@ -745,38 +812,41 @@ def __init__(
         self._sql = sql
         self._result: Optional[TrinoResult] = None
         self._legacy_primitive_types = legacy_primitive_types
-        self._row_mapper: Optional[RowMapper] = None
+        self._row_mapper: Optional[Union[RowMapper, NoOpRowMapper]] = None
```

Comment:
instead of a union we should define a common super type. That has the benefit that in future if we change RowMapper we also are reminded to change NoOpRowMapper as well.

Reply:
Defined common super type `RowMapperBase` [here](https://github.com/trinodb/trino-python-client/blob/3de613f7314803e42544fd886a73b9212e6a3097/trino/client.py#L1220)